src/screens/SettingsScreen.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  StyleSheet,
+  Alert,
+  ActivityIndicator,
+  Image
+} from 'react-native';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { NavigationService } from '../utils/NavigationService';
+import { Storage } from '../utils/Storage';
+import { AccountManager } from '../account/AccountManager';
+import { RootStackParamList } from '../../App';
+import { headerStyles } from '../styles/HeaderStyles';
+
+type SettingsScreenProps = NativeStackScreenProps<RootStackParamList, 'Settings'>;
+
+export const SettingsScreen: React.FC<SettingsScreenProps> = ({ navigation }) => {
+  const [isSignedIn, setIsSignedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    checkAuthStatus();
+  }, []);
+
+  const checkAuthStatus = async () => {
+    const authToken = await Storage.getAuthToken();
+    setIsSignedIn(Boolean(authToken));
+  };
```

Comment:
We can use useCallback here to avoid creating same checkAuthStatus multiple times on every render whaen state changes

const checkAuthStatus = useCallback(async () => {
  const authToken = await Storage.getAuthToken();
  setIsSignedIn(Boolean(authToken));
}, []);

useEffect(() => {
  checkAuthStatus();
}, [checkAuthStatus]);

Reply:
Claude's response to the above solution

```
Actually, in this case using useCallback with useEffect like that would create an infinite loop. Here's why:

When you make checkAuthStatus a useCallback function and add it to the dependency array of useEffect, you're telling React to re-run the effect whenever checkAuthStatus changes
However, since checkAuthStatus uses setIsSignedIn, which is a state setter created by useState, it technically has a dependency that can change between renders
This could create a cycle where the effect runs -> state updates -> component rerenders -> new callback created -> effect runs again
```

Reply:
DONE, I have updated it as Claude suggested, pls recheck.